src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +88,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
+    return cashay.mutate('createUserPicturePutUrl', {
+      variables: {
+        userFilename: pictureFile.name
+      }
+    })
+    .then(({data, error}) => {
+      if (error) {
+        throw new Error(error._error); // eslint-disable-line no-underscore-dangle
+      }
+      return data.createUserPicturePutUrl.picturePutUrl;
+    })
+    .then(picturePutUrl => {
+      return fetch(picturePutUrl, {
```

Comment:
Not exactly. Bob requests a pre-signed PUT URL from the server. If Bob gives this URL to Alice, Alice _could_ use it but only for 60 seconds (by default).

See: http://docs.aws.amazon.com/AmazonS3/latest/dev/PresignedUrlUploadObject.html

Eve would have to be able to decrypt the SSL stream to use this URL.